src/types.ts

```diff
@@ -2063,3 +2063,63 @@ export type User<UserType = UnknownType> = UserType & {
   teams?: string[];
   username?: string;
 };
+
+export type SegmentData = {
+  description: string;
+  // TODO: define this type in more detail
+  filter: {
+    channel?: object;
+    user?: object;
+  };
+  name: string;
+};
+
+export type Segment = {
+  app_pk: number;
+  created_at: string;
+  id: string;
+  updated_at: string;
+  recipients?: number;
+} & SegmentData;
+
+export type CampaignData = {
+  attachments: Attachment[];
+  defaults: Record<string, string>;
+  segment_id: string;
+  text: string;
+  push_notifications?: boolean;
+  sender_id?: string;
+  track_opened?: boolean;
+};
+
+export type CampaignStatus = {
+  errors: string[];
+  status:
+    | 'draft'
+    | 'stopped'
+    | 'scheduled'
+    | 'completed'
+    | 'failed'
+    | 'canceled'
+    | 'in_progress';
+  completed_at?: string;
+  failed_at?: string;
+  progress?: {
```

Comment:
this isn't aligned with api either, we might want to have this way, more user friendly